app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -111,17 +128,25 @@ export class FileViewer extends React.Component {
       this.remove();
     });
 
-    this.setState({loading: true, url: null}, () => {
+    this.setState({loading: true, size: 0, url: null}, () => {
       if (!this.props.selectedFileURL) {
         fetch(
           Routes.get_file_course_submission_path(this.props.course_id, this.props.submission_id, {
             submission_file_id: submission_file_id,
             force_text: force_text,
+            max_size: this.max_size,
           }),
           {credentials: "include"}
         )
           .then(res => res.json())
           .then(body => {
+            this.setState({size: body.size});
```

Comment:
This call isn't synchronous, so you can't assume `this.state.size` will be updated when the `isFileTooLarge` method is called below.

While you could fix this by passing the rest of the code as a callback to `setState`, that's a bit clunky given the volume of that code. Instead, I'll suggest that you make `isFileTooLarge` take an optional argument `size` that, when provided, is used to do the size comparison instead of `this.state.size`. You can then update the state in the subsequent `setState` calls below.